runtime/compiler/runtime/RelocationRecord.cpp

```diff
@@ -2698,66 +2709,75 @@ TR_RelocationRecordProfiledInlinedMethod::preparePrivateData(TR_RelocationRuntim
-         RELO_LOG(reloRuntime->reloLogger(), 1,"\tpreparePrivateData: classChainIdentifyingLoader %p\n", classChainIdentifyingLoader);
-         RELO_LOG(reloRuntime->reloLogger(), 1,"\tpreparePrivateData: classLoader %p\n", classLoader);
-         }
-      J9JavaVM *javaVM = reloRuntime->jitConfig()->javaVM;
-      J9VMThread *vmThread = reloRuntime->currentThread();
-
-      TR_OpaqueClassBlock *inlinedCodeClass;
+      RELO_LOG(reloRuntime->reloLogger(), 6,"\tpreparePrivateData: classLoader %p\n", classLoader);
 
+      if (classLoader)
          {
          TR::VMAccessCriticalSection preparePrivateDataCriticalSection(reloRuntime->fej9());
-         inlinedCodeClass = (TR_OpaqueClassBlock *) jitGetClassInClassloaderFromUTF8(vmThread,
+         inlinedCodeClass = (TR_OpaqueClassBlock *) jitGetClassInClassloaderFromUTF8(reloRuntime->currentThread(),
                                                                                      classLoader,
                                                                                      J9UTF8_DATA(inlinedCodeClassName),
                                                                                      J9UTF8_LENGTH(inlinedCodeClassName));
          }
+      }
+
+   if (inlinedCodeClass && checkInlinedClassValidity(reloRuntime, inlinedCodeClass))
+      {
+      RELO_LOG(reloRuntime->reloLogger(), 6,"\tpreparePrivateData: inlined class valid\n");
+      reloPrivateData->_inlinedCodeClass = inlinedCodeClass;
 
-      if (inlinedCodeClass && checkInlinedClassValidity(reloRuntime, inlinedCodeClass))
+      bool inlinedSiteIsValid = true;
+      TR_OpaqueMethodBlock *inlinedMethod = NULL;
+
+      uintptrj_t *chainData = (uintptrj_t *) reloRuntime->fej9()->sharedCache()->pointerFromOffsetInSharedCache((void *) classChainForInlinedMethod(reloTarget));
+      if (!reloRuntime->fej9()->sharedCache()->classMatchesCachedVersion(inlinedCodeClass, chainData))
+         inlinedSiteIsValid = false;
+
+      if (inlinedSiteIsValid)
          {
-         RELO_LOG(reloRuntime->reloLogger(), 6,"\tpreparePrivateData: inlined class valid\n");
-         reloPrivateData->_inlinedCodeClass = inlinedCodeClass;
-         uintptrj_t *chainData = (uintptrj_t *) reloRuntime->fej9()->sharedCache()->pointerFromOffsetInSharedCache((void *) classChainForInlinedMethod(reloTarget));
-         if (reloRuntime->fej9()->sharedCache()->classMatchesCachedVersion(inlinedCodeClass, chainData))
-            {
-            RELO_LOG(reloRuntime->reloLogger(), 6,"\tpreparePrivateData: classes match\n");
-            TR_OpaqueMethodBlock *inlinedMethod = * (TR_OpaqueMethodBlock **) (((uint8_t *)reloPrivateData->_inlinedCodeClass) + vTableSlot(reloTarget));
+         inlinedMethod = * (TR_OpaqueMethodBlock **) (((uint8_t *)reloPrivateData->_inlinedCodeClass) + vTableSlot(reloTarget));
+         if (!inlinedMethod)
+            inlinedSiteIsValid = false;
```

Comment:
I don't think it's possible for the vtable slot to be null

Reply:
I see, do you think it's fine as is, or would you prefer I rebase with this check removed?

Reply:
I'd ask for it to be changed if there was something else to necessitate a round of updates, but I think it can be left alone. It doesn't really hurt